packages/eslint-plugin/lib/rules/react-require-autocomplete.js

```diff
@@ -0,0 +1,137 @@
+const {findDefinition} = require('../utilities');
+
+const autocompleteInputTypes = [
+  'color',
+  'date',
+  'datetime-local',
+  'email',
+  'month',
+  'number',
+  'password',
+  'range',
+  'search',
+  'tel',
+  'text',
+  'time',
+  'url',
+  'week',
+];
+const autocompleteElementNames = ['input', 'TextField'];
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Require autocomplete attribute on certain input types.',
+      category: 'Best Practices',
+      recommended: true,
+    },
+    schema: [],
+  },
+  create: (context) => ({
+    JSXOpeningElement: (node) => {
+      if (!autocompleteElementNames.includes(node.name.name)) {
+        return;
+      }
+
+      // input elements without a type attribute are treated as text
+      // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#input_types
+      let [inputType] = findAttrValue(node.attributes, 'type', context);
+      inputType = inputType || 'text';
+
+      if (!autocompleteInputTypes.includes(inputType)) {
+        return;
+      }
+
+      const [autocompleteAttribute, completeCheck] = findAttrValue(
+        node.attributes,
+        'autoComplete',
+        context,
+      );
+
+      // completeCheck may be false if the element has props passed in
+      // with object spread syntax because we might not be be able to
+      // find the variable definition in the tree. In this case we do
+      // not report an error
+      if (autocompleteAttribute || !completeCheck) {
+        return;
+      }
+
+      context.report(
+        node,
+        `Input elements of type '${inputType}' must have an autoComplete attribute`,
+      );
+    },
+  }),
+};
+
+function findAttrValue(attributes, name, context) {
```

Comment:
Interesting, I’d recommend using a module like [jsx-ast-utils](https://www.npmjs.com/package/jsx-ast-utils) so we don’t have to implement these utils ourselves?